### PR TITLE
Obliterate branches in delete.namespace rather than consing an empty branch onto existing history.

### PR DESCRIPTION
**Pull request by ChrisPenner**
~EDIT: seems like this change doesn't actually work and needs a bit more looking into iff we decide to go forward with this plan~

Update: Okay took the time to do it correctly, updated the transcripts as well.

## Overview

Currently, deleting a namespace doesn't actually delete the history, it just conses an empty branch onto the history at that location. The distinction is important, and might be relevant in certain cases, but I think @stew and @ceedubs  have proven through experience that the current behaviour is rarely what's desired.

This PR changes the behaviour so that the history at the delete path is empty after deletion. If users want to revive the old branch it will still exist in the history of the parent namespace.

## Implementation

* Change the delete operation to replace the branch at the specified path with an empty branch (with an empty history) within the parent branch. 
* Change the delete root operation to just replace the namespace root with an empty causal instead of stepping forward in history with an empty branch.

## Testing

Updated the altered transcripts and confirmed they're behaving as expected.

### CODE REVIEW

**Comment by pchiusano** [comment]
So as I understand it, the thinking behind the original "don't obliterate" behavior was that if Alice deletes the `foo` subnamespace of a project, and Bob adds the definition `foo.bar.baz`, and they merge, we wanted a merge result where only `foo.bar.baz` remains. Whereas if you obliterate the history of `foo`, you might end up with all of `foo` getting resurrected by the merge.

However, in practice, keeping the history around has resulted in footguns, where you are unwittingly merging merging with stray "deleted namespace history" without even being aware this is happening. And the results are incredibly confusing and easy to not notice until later.

Going back to the "merges resurrecting subnamespace" behavior, it's not even clear to me that "don't resurrect" is always preferred anyway. Definitions within a namespace often have mutual dependencies, so a merge that results in `foo.bar.baz` only and deletes the rest might not be self-contained anymore.

So in conclusion, we're now thinking that deleting a namespace should just delete it from its parent namespace, so that the parent namespace literally has no child for that key.

**Comment by aryairani** [comment]
I want to workshop our understanding of namespace manipulation and merges, produce a diagram and a short doc, and use it to guide implementation and bugfixes going forward. I will create a separate ticket to this effect.

**Comment by pchiusano** [comment]
This continues to be a footgun and it's very easy to run into it during a library upgrade process. I think we should just do the obliterating; it's an improvement over the current weird behavior, and doesn't preclude us giving this a more serious rethink at our leisure.

File: unison-cli/src/Unison/Codebase/Editor/HandleInput.hs [hunk 1]
```diff
@@ -911,8 +909,12 @@ loop e = do
...
                       Cli.respondNumbered $ CantDeleteNamespace ppeDecl endangerments
                       Cli.returnEarlyWithoutOutput
-                Cli.stepAt description $
-                  BranchUtil.makeDeleteBranch (Path.convert absPath)
+                parentPathAbs <- Cli.resolvePath' parentPath
+                -- We have to modify the parent in order to also wipe out the history at the
+                -- child.
+                Cli.updateAt description parentPathAbs \parentBranch ->
+                  parentBranch
+                    & Branch.modifyAt (Path.singleton childName) \_ -> Branch.empty
```
Comment (pchiusano) [inline unison-cli/src/Unison/Codebase/Editor/HandleInput.hs:917]: Right, that makes sense.

**Review by pchiusano — APPROVED** [review]

**Review by pchiusano — APPROVED** [review]

File: unison-cli/src/Unison/Codebase/Editor/HandleInput.hs [hunk 2]
```diff
@@ -911,8 +909,12 @@ loop e = do
                       ppeDecl <- currentPrettyPrintEnvDecl Backend.Within
                       Cli.respondNumbered $ CantDeleteNamespace ppeDecl endangerments
                       Cli.returnEarlyWithoutOutput
-                Cli.stepAt description $
-                  BranchUtil.makeDeleteBranch (Path.convert absPath)
```
Comment (mitchellwrosen) [inline unison-cli/src/Unison/Codebase/Editor/HandleInput.hs:915]: `BranchUtil.makeDeleteBranch` is now unused, let's delete it

**Review by mitchellwrosen — APPROVED** [review]

**Review by tstat — APPROVED** [review]
